clang/lib/Driver/Driver.cpp

```diff
@@ -2230,6 +2239,7 @@ bool Driver::HandleImmediateArgs(const Compilation &C) {
   }
 
   if (C.getArgs().hasArg(options::OPT_print_runtime_dir)) {
+    initDarwinTarget();
```

Comment:
We should also add tests for this behavior. We were not printing the right directory in the embedded case before your patch, but we were printing the right path in the non-embedded case. I don't know if your patch fixes it. If your patch doesn't fix the embedded case, we could scale down the patch a bit and avoid touching the `OPT_print_runtime_dir` code path (like you did initially) just for the sake of landing this. The incorrect `-print-runtime-dir` result on embedded machO targets could be fixed separately.

Reply:
I removed this change for now because I think we should probably fix this separately. Only adding the early init did not change the behavior for the cases we are interested in.